feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/PreviewViewModel.kt

```diff
@@ -550,14 +562,34 @@ class PreviewViewModel @AssistedInject constructor(
                         }
                     }, contentResolver, imageCaptureUri, ignoreUri).savedUri
                 },
-                onSuccess = { savedUri -> onImageCapture(ImageCaptureEvent.ImageSaved(savedUri)) },
+                onSuccess = { savedUri ->
+                    onImageCapture(ImageCaptureEvent.ImageSaved(savedUri), uriIndex)
+                    incrementExternalMultipleImageCaptureModeUriIndexIfNeeded()
```

Comment:
Should this be incremented inline with `captureImageInternal`? If you wait for the success/failure callback, you could potentially create multiple image capture requests for the same URI if you press the capture button fast enough.